dev-client/src/model/soilId/actions/localSoilDataActions.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {cloneDeep} from 'lodash';
+
+import {
+  DepthDependentSoilDataUpdateMutationInput,
+  DepthInterval,
+  SoilDataDeleteDepthIntervalMutationInput,
+  SoilDataUpdateDepthIntervalMutationInput,
+  SoilDataUpdateMutationInput,
+} from 'terraso-client-shared/graphqlSchema/graphql';
+import {SoilData} from 'terraso-client-shared/soilId/soilIdTypes';
+
+import {
+  DEPTH_DEPENDENT_SOIL_DATA_UPDATE_FIELDS,
+  DEPTH_INTERVAL_UPDATE_FIELDS,
+  SOIL_DATA_UPDATE_FIELDS,
+} from 'terraso-mobile-client/model/soilId/actions/soilDataActionFields';
+import {DEFAULT_DEPTH_INTERVAL_PRESET} from 'terraso-mobile-client/model/soilId/soilDataConstants';
+import {
+  compareInterval,
+  sameDepth,
+} from 'terraso-mobile-client/model/soilId/soilIdFunctions';
+
+export const updateSoilData = (
+  input: SoilDataUpdateMutationInput,
+  data?: SoilData,
+): SoilData => {
+  const result = initializeResult(data);
+  updateFields(input, result, SOIL_DATA_UPDATE_FIELDS);
+
+  /* Specifying a depth interval preset clears all existing depth intervals */
```

Comment:
Nit: Could be helpful to point to where the corresponding backend logic lives

Reply:
Good idea, will do